### DISCUSSION TITLE
Allow sender to earn the interest on the principal locked up in the stream

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/evm-monorepo#830] [Ideas]
## Problem

- Company creates stream for employee
- Salary has fixed value
- Company wants to stream the fixed value to the employee
- Company also wants to earn a yield on the idle capital not streamed to the employee already
- Company tries to use an ERC-4626 vault but the yield goes to the recipient

## Solution

There's no simple solution, unfortunately. Retaining the yield for the sender involves a redesign of the core contracts.

- The naive solution is to have bespoke integrations for yield protocols, e.g. [DSR](https://docs.makerdao.com/smart-contract-modules/proxy-module/dsr-manager-detailed-documentation)
- The proper solution is a modular plugin system that can be used for any yield protocol

That said, I'm skeptical that this is worth implementing when simpler alternatives exist - for example, we could use Gelato's [Web3 Functions](https://gelato.network/blog/web3-functions-serverless-apps) to automate the creation of streams.

## Related

An important discussion that is related: [Alternative source of revenues: lending user deposits](https://github.com/sablier-labs/company-discussions/discussions/12)

**Comment by PaulRBerg** [comment]
[Kenton Prescott](https://twitter.com/KentonPrescott)'s feature request:

> I'd use this if the destination addresses can be obfuscated / made private

**Comment by PaulRBerg** [comment]
This is perhaps the most requested feature to date (since we launched V2).

Aragon, Gelato, Spark, and other organizations we talked to have all requested the ability to earn a yield on top of the principal getting streamed.

**Comment by PaulRBerg** [comment]
It may be helpful to explore how Sushi has enabled this feature in [Furo](https://twitter.com/bantg/status/1532862087306919936).

**Comment by razgraf** [comment]
I don't think they did though. At least not in the way we need it.

[Looks](https://www.sushi.com/academy/articles/introducing-furo) like interest is earned by the recipient (not the sender) by simply holding the funds in Bento while they're being streamed and only moving them out at withdrawal. They ask you to choose the source of those funds, but I'm curious if they still move them in Bento if you choose the wallet option.

<img width="364" alt="Screenshot 2023-09-15 at 17 12 43" src="https://github.com/sablier-labs/v2-core/assets/15740806/aeec90ae-4b74-46da-837e-880c40af2075">

Anyway, impossible to research further because I cannot seem to find their contracts ...anywhere.

**Comment by PaulRBerg** [comment]
hah, thanks for taking a look, @razgraf! That's interesting.

Based on the technical difficulties unsurfaced in your conversation yesterday, and on the present state of ERC-4626 vaults ([~20% of implementations don't adhere to the standard](https://medium.com/@vikram.arun/multichain-erc-4626-conformance-f34b682b273b)), I suggest postponing this feature until the community standards mature.

**Comment by canny** [comment]
This issue has been linked to a Canny post: [Allow sender to earn the interest on the principal locked up in the stream](https://sablier.canny.io/admin/board/feature-requests/p/allow-sender-to-earn-the-interest-on-the-principal-locked-up-in-the-stream) :tada:

**Comment by smol-ninja** [comment]
Given the lack of clarity regarding the deliverable, should we close this issue for now and start a discussion out of it?
